invenio_admin/views.py

```diff
@@ -22,24 +22,44 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Administration interface for Invenio."""
+"""Admin view class factory for creating protected admin views on-the-fly."""
 
 from __future__ import absolute_import, print_function
 
-from flask_admin import AdminIndexView
+from flask import current_app, redirect, request, url_for
 from flask_login import current_user
+from werkzeug.local import LocalProxy
+
+current_admin = LocalProxy(lambda: current_app.extensions['invenio-admin'])
 
 
 def protected_adminview_factory(cls):
-    """Factory for creating protected view classes."""
+    """Factory for creating protected admin view classes.
+
+    The factory will create a new class using the provided class as base class
+    and overwriting ``is_accessible()`` and ``inaccessible_callback()``
+    methods.
+
+    :param cls: Class to use as base class.
```

Comment:
it would be nice to rename it to `base_cls` or `base_class` so it doesn't collide later with `@classmethod` argument `cls`.